### PR TITLE
Option to generate classes instead of case classes when fields.size > 22

### PR DESCRIPTION
**Pull request by mariussoutier**
Added parameter `restrictedFieldNumber` to Generator so classes are generated instead of case classes when parameter is true AND record has more than 22 fields.

Right now I'm checking for 22 fields explicitly, so it's not very generic, maybe that can be changed in the future.

### CODE REVIEW

**Comment by mariussoutier** [comment]
Fixes #42 .

File: avrohugger-core/src/main/scala/Generator.scala [hunk 1]
```diff
@@ -41,58 +42,65 @@ class Generator(format: SourceFormat,
   //////////////// methods for writing definitions out to file /////////////////
   def schemaToFile(
     schema: Schema,
-    outDir: String = defaultOutputDir): Unit = {
+    outDir: String = defaultOutputDir,
+    restrictedFields: Boolean = false): Unit = {
```
Comment (julianpeeters) [inline avrohugger-core/src/main/scala/Generator.scala]: Looking great mostly. I assume this was meant to be like you did the other methods, with the resitrictedFields commented out? In this case I'm in favor of using the global restrictedFieldNumber in order to keep the api slim and unchanging across versions if we can help it.

And while these methods are covered in various tests, the above inconsistency actually results in a bug where the restrictedFields set in the Generator can be ignored, suggesting tests are needed here too: I think a large (>22) Schema and a large IDL, each in String format and in a File, and each of those toString and toFile (1 avsc and 1 avdl reused in 8 total examples) would help. I'll fix the odd method, remove the commented params, and publish after the tests are added, this weekend if you don't beat me to it -- tests and a note about this feature in the README I guess.

Comment (mariussoutier) [reply avrohugger-core/src/main/scala/Generator.scala]: Ah yes, thanks, good catch.

Comment (mariussoutier) [reply avrohugger-core/src/main/scala/Generator.scala]: I'll definitely add tests, just wanted to get quick feedback if this is the right way to go.

Comment (mariussoutier) [reply avrohugger-core/src/main/scala/Generator.scala]: I updated the PR and removed the unused params.

**Comment by julianpeeters** [comment]
Very cool. I expect that this wil work out well with the sbt plugin. Now I believe it will be as easy as passing in something like `scalaVersion.value == 2.10.X` boolean to the generator in the FileWriter.

**Comment by mariussoutier** [comment]
It's even easier with `scalaBinaryVersion.value == "2.10"`.

**Comment by julianpeeters** [comment]
Oh wow awesome :D

**Comment by mariussoutier** [comment]
Ok, first set of tests incoming... Already fixed some bugs.

**Comment by mariussoutier** [comment]
And the rest...

File: avrohugger-core/src/test/scala/util/Util.scala [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
+  def readFile(fileName: String, maxTries: Int = 3): String = {
+    def readFile0(count: Int): String = {
+      try { // if file is empty, try again, it should be there
+        val contents: String = scala.io.Source.fromFile(fileName).mkString
+        if (contents.isEmpty && (count < maxTries)) readFile0(count + 1)
+        else contents
+      } catch { // if file is not found, try again, it should be there
+        case e: Throwable =>
+          if (count < maxTries) readFile0(count + 1)
+          else sys.error("File not found: " + fileName)
```
Comment (julianpeeters) [inline avrohugger-core/src/test/scala/util/Util.scala:14]: nice

**Review by julianpeeters — APPROVED** [review]

**Pull request merged**